buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ValidateRestSpecPlugin.java

```diff
@@ -39,6 +39,16 @@ public void apply(Project project) {
                 }));
                 task.setJsonSchema(new File(project.getRootDir(), "rest-api-spec/src/main/resources/schema.json"));
             });
-        project.getTasks().named("precommit").configure(t -> t.dependsOn(validateRestSpecTask));
+
+        Provider<ValidateJsonNoKeywordsTask> validateNoKeywordsTask = project.getTasks()
+            .register("validateNoKeywords", ValidateJsonNoKeywordsTask.class, task -> {
+                task.setInputFiles(Util.getJavaTestAndMainSourceResources(project, filter -> {
+                    filter.include(DOUBLE_STAR + "/rest-api-spec/api/" + DOUBLE_STAR + "/*.json");
+                    filter.exclude(DOUBLE_STAR + "/_common.json");
+                }));
+                task.setJsonKeywords(new File(project.getRootDir(), "rest-api-spec/src/main/resources/keywords.json"));
+            });
+
+        project.getTasks().named("precommit").configure(t -> t.dependsOn(validateRestSpecTask).dependsOn(validateNoKeywordsTask));
```

Comment:
Although I guess this "builder" type pattern of chaining `dependsOn` works, I find it a bit odd. Just to placate my neurosis let's just do `dependsOn(validateRestSpecTask, validateNoKeywordsTask)` as I just find it easier to grok what's happening using the vararg method.